packages/react-ui/src/hooks/useImperativeIntl.js

```diff
@@ -9,21 +8,31 @@ import {
 
 const cache = createIntlCache();
 
+let lastIntlConfig;
+let cachedC4rIntl;
+
 export default function useImperativeIntl(intlConfig) {
-  return useMemo(() => {
+  if (!cachedC4rIntl || lastIntlConfig !== intlConfig) {
+    // this is very naive cache that bases on fact that Intl instance is actually same for most of time
+    // so we can reuse those maps across several instances of same components
+    // note, useMemo can't do that globally and flattenMessages over _app_ and c4r is quite costly and would
+    // be paid for evey c4r component mounted
```

Comment:
This cache assumes we have 1 active Intl instance in app which is usually the case.

It _can_ be changed to WeakMap<IntlConfig, IntlShape> if we have use case where one app use many languages at once, not sure if that's the case

Reply:
Changed to WeakMap, so we're on safe side.